fortran/src/cudafor.f90

```diff
@@ -70,45 +70,29 @@ end function cudaSetDevice
 end interface cudaSetDevice
 !End cudaSetDevice
 
-!Start cudaMAlloc
-interface cudaMAlloc
-integer(c_int) function cudaMAlloc(devPtr, size) bind(c, name = "cudaMalloc")
+!Start cudaMalloc
+interface cudaMalloc
+integer(c_int) function cudaMalloc(devPtr, size) bind(c, name = "cudaMalloc")
 import :: c_int, c_size_t
 import :: c_devptr
 implicit none
 type(c_devptr) :: devPtr
 integer(c_size_t), value :: size
-end function cudaMAlloc
-end interface cudaMAlloc
-!End cudaMAlloc
+end function cudaMalloc
+end interface cudaMalloc
+!End cudaMalloc
 
 !Start cudaMemCpy
 interface cudaMemCpy
```

Comment:
`cudaMemcpy` is a generic interface to `cudaMemCpyH2D` and `cudaMemCpyD2H`. Only `cudaMemcpy` is exposed (`public` statement towards the top of the file). `cudaMemCpyH2D` and `cudaMemCpyD2H` differ in the type of their `dst` and `src` arguments. `cudaMemcpy` is an overloaded function and automatically selects `cudaMemCpyH2D` if `dst`/`src` are `c_devptr`/`c_ptr` and `cudaMemCpyD2H` if `dst`/`src` are `c_ptr`/`c_devptr`. As far as I know, this is the standards-compliant way of overloading a Fortran function to allow a parameter to have different types. The way I had it works for GNU-4.9.2+, Intel-16 and PGI-16 (PGI tested by removing the `-Mcuda,cudaX.Y` flags and explicitly compiling cudafor.f90 as done for GNU and Intel), but removing the overloading doesn't work for any of them.